api/pkg/s3/bucketlifecycleput.go

```diff
@@ -116,6 +117,18 @@ func checkValidationOfActions(actions []*s3.Action) error {
 	return nil
 }
 
+func (s *APIService) checkModifyLifecycleConfig(bucketLcCfg *s3.LifecycleRule, rule *s3.LifecycleRule) bool {
```

Comment:
imho, this is too complicated.
an Update for a lifecycle rule, will simply be a put. from the user pov, 

1. i need to first see the current lifecycle rules on the bucket (we can support a getLifecycleRule(bucket))
2. now, i will update it and call the update/put
3. we support an updateLifecycle(LifecycleRule, bucket). S3 documentation says

> Sets lifecycle configuration for your bucket. If a lifecycle configuration exists, it replaces it.

and we are done !

>